Model/Data/AmwalOrderDetails.php

```diff
@@ -117,16 +117,9 @@ private function isPayValid($amwalOrderId)
         }
 
         $orderStates = [
-            'pending_payment',
-            'new',
-            'holded',
-            'canceled',
+            'pending_payment'
         ];
-
-        if (!in_array($orderState, $orderStates)) {
-            return false;
-        }
-        return true;
+        return in_array($orderState, $orderStates);
```

Comment:
Since the only `'pending_payment'` is left in the  `$orderStates` array, we can simply change this to a check without the need of `in_array`:

```suggestion
        return $orderState === 'pending_payment';
```

If you apply it this way, don't forget to remove the `$orderStates` array

Reply:
You are right, but I left it for the possibility that another case will be added in the future.